src/state.rs

```diff
@@ -69,6 +72,14 @@ impl State {
 		match *event {
 			Event::Ready(ref ready) => *self = State::new(ready.clone()),
 			Event::UserUpdate(ref user) => self.user = user.clone(),
+			Event::UserNoteUpdate(ref user_id, ref note) => {
+				match self.notes.as_mut() {
+					Some(notes) => {
+						notes.insert(UserId(user_id.0), note.clone());
+					},
+					None => {},
```

Comment:
An if-let rather than a `match` would reduce nesting here, but it's a pretty minor stylistic concern.